gateway.go

```diff
@@ -100,6 +100,9 @@ func (s *gatewayResource) gatewayHandler(w http.ResponseWriter, r *http.Request)
 		if err == ConfigMismatchError {
 			http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
 			return
+		} else if err == TargetForbiddenError {
```

Comment:
This change shouldn't be necessary since`TargetForbiddenError` will always be encapsulated.

Reply:
I don't think this is true. This error is always encapsulated and only known to the client.